utils/gen-snapshot-tests.js

```diff
@@ -95,41 +96,100 @@ function generateTestFile(root, examples, componentFileMap) {
     console.log(`wrote ${outPath}`);
 }
 
-for (const section of styleguideConfig.sections) {
-    const content = fs.readFileSync(section.content, "utf8");
+function readExamplesFromDocument(documentPath) {
+    if (!fs.existsSync(documentPath)) {
+        return [];
+    }
+    const content = fs.readFileSync(documentPath, "utf8");
     const tokens = marked.lexer(content);
     const examples = tokens
         .filter((token) => token.type === "code")
         .map((token) => token.text);
-    if (examples.length === 0) {
-        // eslint-disable-next-line no-console
-        console.warn(`no examples for section ${section.name}`);
-        continue;
+
+    return examples;
+}
```

Comment:
Helper function so we can reuse it. All it does is grab examples. Also encapsulates checking that the path is valid, just for ease of use.

Reply:
Nice!